client/src/components/app.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import { getProducts } from '../actions/index';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Product from './product';
+import Search from './search';
+
+class App extends Component {
+  // get products on load
+  componentDidMount() {
+    this.props.getProducts();
+  }
+
+  // get page and query 
+  getPage = (e) => {
+    const page = e.target.innerHTML;
+    this.props.getProducts({ ...Search.WrappedComponent.values, page });
```

Comment:
Hmm... I'm honestly not sure what I think of it. It's clever, but I'm wondering if this requires that `App` know too much about how `Search` is actually `Search` wrapped in a special `reduxForm` component. Perhaps we might want to keep these "values" somewhere else in redux instead of having to pick them off a component?